arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -1351,6 +1367,109 @@ void RocksDBEngine::prepareDropCollection(TRI_vocbase_t& /*vocbase*/,
   replicationManager()->drop(&coll);
 }
 
+arangodb::Result RocksDBEngine::dropCollectionFast(TRI_vocbase_t& vocbase,
+                                                   LogicalCollection& collection) {
+
+  auto* rocksdb = static_cast<RocksDBMetaCollection*>(collection.getPhysical());
+  bool const compactAfterDrop = rocksdb->meta().numberDocuments() >= 32 * 1024;
+
+  rocksdb::DB* db = _db->GetRootDB();
+
+  TRI_ASSERT(collection.status() == TRI_VOC_COL_STATUS_DELETED);
+
+  // Prepare collection remove batch
+  rocksdb::WriteBatch batch;
+  {
+    RocksDBLogValue logValue =
+        RocksDBLogValue::CollectionDrop(vocbase.id(), collection.id(),
+                                        arangodb::velocypack::StringRef(
+                                            collection.guid()));
+    batch.PutLogData(logValue.slice());
+  }
+  {
+    RocksDBKey key;
+    key.constructCollection(vocbase.id(), collection.id());
+    batch.Delete(RocksDBColumnFamilyManager::get(RocksDBColumnFamilyManager::Family::Definitions),
```

Comment:
Not sure about the reasons for adding a `Delete` to a batch going wrong, but at least `Delete` returns a `Status` which could indicate an error. So we should check it.

Reply:
By the way, the same seems to be true for `PutLogData`. So if we are changing the code, let's fix the error checking as well.